app/src/main/java/pl/animagia/FullscreenPlaybackActivity.java

```diff
@@ -304,8 +322,110 @@ protected void onDestroy() {
         playerRestarter = null;
     }
 
+    private void initSpinnerOnTouch(boolean subtitleOnSelected){
+        spinnerOfQuality = findViewById(R.id.spinner_quality);
+        String[] quality = getResources().getStringArray(R.array.quality);
+        ArrayAdapter<String> adapterQuality = new ArrayAdapter<>(
+                this, R.layout.spinner_item, quality
+        );
+
+        adapterQuality.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+
+        spinnerOfQuality.setAdapter(adapterQuality);
+        spinnerOfQuality.setOnTouchListener(new View.OnTouchListener(){
+
+            @Override
+            public boolean onTouch(View view, MotionEvent motionEvent) {
+                final Toast t = Toast.makeText(context, "Nie ma możliwości zmian jakości w preview", Toast.LENGTH_SHORT);
```

Comment:
Show a dialog, not a toast, and only when motionEvent.getAction() == MotionEvent.ACTION_UP (or you'll end up with 3 dialogs).